src/state_circuit.rs

```diff
@@ -1,3 +1,3 @@
 //! The state circuit implementation.
 
-pub(crate) mod memory;
+pub(crate) mod state;
```

Comment:
Do we expect another `mod storage`?

Reply:
Not sure right now, I plan to experiment with this in #24. I might start there with all checks for stack and memory in one file and then modularize it (storage would probably come in a later PR). Would be that ok?